hare3/metrics.go

```diff
@@ -0,0 +1,67 @@
+package hare3
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/spacemeshos/go-spacemesh/metrics"
+)
+
+const namespace = "hare"
+
+var (
+	processCounter = metrics.NewCounter(
+		"session",
+		namespace,
+		"number of hare sessions at different stages",
+		[]string{"stage"},
+	)
+	sessionStart      = processCounter.WithLabelValues("started")
+	sessionTerminated = processCounter.WithLabelValues("terminated")
+	sessionCoin       = processCounter.WithLabelValues("weakcoin")
+	sessionResult     = processCounter.WithLabelValues("result")
+
+	exitErrors = metrics.NewCounter(
+		"exit_errors",
+		namespace,
+		"number of unexpected exit errors. should remain at zero",
+		[]string{},
+	).WithLabelValues()
+	validationError = metrics.NewCounter(
+		"validation_errors",
+		namespace,
+		"number of validation errors. not expected to be at zero",
+		[]string{"error"},
+	)
+	notRegisteredError = validationError.WithLabelValues("not_registered")
+	malformedError     = validationError.WithLabelValues("malformed")
+	signatureError     = validationError.WithLabelValues("signature")
+	oracleError        = validationError.WithLabelValues("oracle")
+	maliciuosError     = validationError.WithLabelValues("malicious")
+
+	droppedMessages = metrics.NewCounter(
+		"dropped_msgs",
+		namespace,
+		"number of messages dropped by gossip",
+		[]string{},
+	).WithLabelValues()
+
+	validationLatency = metrics.NewHistogramWithBuckets(
+		"validation_seconds",
```

Comment:
in case of subsecond it will be displayed as normal float 0.001 for example. nanoseconds are quite hard to see, with fraction of seconds is more usable for me

Reply:
actually another reason is what you are using as lowest boundary. i thihk that anything that takes less than 1ms is negligible in such protocol (currently lowest boundary is 0.5s)